src/org/labkey/test/pages/core/admin/LimitActiveUserPage.java

```diff
@@ -0,0 +1,115 @@
+package org.labkey.test.pages.core.admin;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.pages.LabKeyPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+
+public class LimitActiveUserPage extends LabKeyPage<LimitActiveUserPage.ElementCache>
+{
+    public LimitActiveUserPage(WebDriver driver)
+    {
+        super(driver);
+        waitForPage();
+    }
+
+    public static LimitActiveUserPage beginAt(WebDriverWrapper wrapper)
+    {
+        wrapper.beginAt(WebTestHelper.buildURL("user", "limitActiveUsers"));
+        return new LimitActiveUserPage(wrapper.getDriver());
+    }
+
+    public LimitActiveUserPage userWarning(String value)
+    {
+        elementCache().userWarning.selectByVisibleText(value);
+        return this;
+    }
+
+    public LimitActiveUserPage limitActiveUsers(String value)
+    {
+        elementCache().limitActiveUsers.selectByVisibleText(value);
+        return this;
+    }
+
+    public String getUserWarningLevel()
+    {
+        return elementCache().userWarningLevel.get();
+    }
+
+    public LimitActiveUserPage setUserWarningLevel(String value)
+    {
+        elementCache().userWarningLevel.set(value);
+        return this;
+    }
+
+    public String getUserLimitLevel()
+    {
+        return elementCache().userLimitLevel.get();
+    }
+
+    public LimitActiveUserPage setUserLimitLevel(String value)
+    {
+        elementCache().userLimitLevel.set(value);
+        return this;
+    }
+
+    public String getUserWarningMessage()
+    {
+        return elementCache().userWarningMessage.getText();
+    }
+
+    public LimitActiveUserPage setUserWarningMessage(String value)
+    {
+        elementCache().userWarningMessage.clear();
+        elementCache().userWarningMessage.sendKeys(value);
+        return this;
+    }
+
+    public String getUserLimitMessage()
+    {
+        return elementCache().userLimitMessage.getText();
+    }
+
+    public LimitActiveUserPage setUserLimitMessage(String value)
+    {
+        elementCache().userLimitMessage.clear();
+        elementCache().userLimitMessage.sendKeys(value);
+        return this;
+    }
+
+    public ShowAdminPage save()
+    {
+        clickAndWait(elementCache().saveBtn);
+        return new ShowAdminPage(getDriver());
+    }
```

Comment:
You may want to add a `saveExpectingError` method. Since the limit values are just text boxes they can be set to anything including invalid values. _Hint: you may want to test for that._

And since they can be set to invalid value you may need to add a getter for any error messages that are shown.

Reply:
Good catch.

Reply:
Done